packages/@tinacms/graphql/src/database/index.ts

```diff
@@ -211,27 +212,32 @@ export class Database {
     data: { [key: string]: unknown },
     collection?: string
   ) => {
-    if (SYSTEM_FILES.includes(filepath)) {
-      throw new Error(`Unexpected put for config file ${filepath}`)
-    } else {
-      let collectionIndexDefinitions
-      if (collection) {
-        const indexDefinitions = await this.getIndexDefinitions()
-        collectionIndexDefinitions = indexDefinitions?.[collection]
-      }
+    try {
+      if (SYSTEM_FILES.includes(filepath)) {
+        throw new Error(`Unexpected put for config file ${filepath}`)
+      } else {
+        let collectionIndexDefinitions
+        if (collection) {
+          const indexDefinitions = await this.getIndexDefinitions()
+          collectionIndexDefinitions = indexDefinitions?.[collection]
+        }
 
-      const { stringifiedFile, payload, keepTemplateKey } =
-        await this.stringifyFile(filepath, data)
-      if (this.store.supportsSeeding()) {
-        await this.bridge.put(normalizePath(filepath), stringifiedFile)
+        const { stringifiedFile, payload, keepTemplateKey } =
+          await this.stringifyFile(filepath, data)
+        if (this.store.supportsSeeding()) {
+          await this.bridge.put(normalizePath(filepath), stringifiedFile)
+        }
+        await this.store.put(normalizePath(filepath), payload, {
+          keepTemplateKey,
+          collection: collection,
+          indexDefinitions: collectionIndexDefinitions,
+        })
       }
-      await this.store.put(normalizePath(filepath), payload, {
-        keepTemplateKey,
-        collection: collection,
-        indexDefinitions: collectionIndexDefinitions,
-      })
+      return true
+    } catch (error) {
+      console.error(`Error in PUT for ${filepath}`)
+      throw error
```

Comment:
Is `console.log` the primary way of communicating an error happened? I think ideally the error message speaks for itself, so in this scenario wed' throw a new error with a more clear message than the one we're catching, and then not also log the error.

Reply:
For sure! Fixed.